packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -769,19 +769,17 @@ class AppInstance {
 
 /// A [Logger] which sends log messages to a listening daemon client.
```

Comment:
Would it make sense to have two subclasses of _AppRunLogger, one for when you want to send all events to the IDE, and one for when you only want to send progress events to the IDE but want status, error, and trace events to instead be logged to the console?

I still don't really understand what this is for.

The actual change (from logToStdout to having an explicit parent) seems like an improvement, I'm just stuck on the underlying code that you're changing here.